### PR TITLE
Task/10/verify signature

### PR DESCRIPTION
**Pull request by aleixmorgadas**
@aizaiz, you might be interested in the changes I made.

- Added the method `signData` to `Account` class
- Changed `verify` method from static to instance method of `PublicAccount`

### CODE REVIEW

File: src/model/account/PublicAccount.ts [hunk 1]
```diff
@@ -53,6 +54,43 @@ export class PublicAccount {
...
+    /**
+     * Verify a signature.
+     *
+     * @param {string} data - The data to verify.
+     * @param {string} signature - The signature to verify.
+     *
+     * @return {boolean}  - True if the signature is valid, false otherwise.
+     */
+    public verifySignature(data: string, signature: string): boolean {
+        if (!data || !signature) {
```
Comment (Vektrat) [inline src/model/account/PublicAccount.ts]: is it possible there is a case where the signed data is ""? (no data)

File: src/model/account/PublicAccount.ts [hunk 2]
```diff
@@ -53,6 +54,43 @@ export class PublicAccount {
...
+            throw new Error('Missing argument');
+        }
+
+        if (signature.length !== 128) {
```
Comment (Vektrat) [inline src/model/account/PublicAccount.ts]: can we use a reference to some declaration of what this size actually is?

File: src/model/account/PublicAccount.ts [hunk 3]
```diff
@@ -53,6 +54,43 @@ export class PublicAccount {
...
+        if (!convert.isHexString(signature)) {
+            throw new Error('Signature must be hexadecimal only');
+        }
+
+        // Convert signature key to Uint8Array
+        const convertedSignature = convert.hexToUint8(signature);
+
+        let convertedData;
+
+        // Convert data to hex if data is not hex
```
Comment (Vektrat) [inline src/model/account/PublicAccount.ts]: I made this comment in the previous specific PR, are we sure we want to dynamically parse types in this function? Also, maybe the function documentation should specify the expected data format

File: test/model/account/Account.spec.ts [hunk 4]
```diff
@@ -45,4 +46,14 @@ describe('Account', () => {
...
 
+    it('signData', () => {
+        const account = Account.createFromPrivateKey(
+            'AB860ED1FE7C91C02F79C02225DAC708D7BD13369877C1F59E678CC587658C47',
+            NetworkType.MIJIN_TEST,
+        );
+        const publicAccount = account.publicAccount;
+        const signed = account.signData('catapult rocks!');
+        expect(publicAccount.verifySignature('catapult rocks!', signed))
+            .to.be.equal(true);
```
Comment (Vektrat) [inline test/model/account/Account.spec.ts]: `to.be.true`

I've seen this repeats in other tests as well ^^

File: test/model/account/PublicAccount.spec.ts [hunk 5]
```diff
@@ -27,3 +27,84 @@ describe('PublicAccount', () => {
...
+
+        // Act & Assert:
+        expect(signerPublicAccount.verifySignature(data, signature)).equal(false);
+    });
+
+    it('Throw error if signature verification is missing a parameter', () => {
+        // Arrange:
+        const signerPublicAccount = PublicAccount.createFromPublicKey('22816F825B4CACEA334723D51297D8582332D8B875A5829908AAE85831ABB508',
+            NetworkType.MIJIN_TEST);
+        const data = '';
```
Comment (Vektrat) [inline test/model/account/PublicAccount.spec.ts]: as commented before, is this an invalid case?

File: src/model/account/PublicAccount.ts (hunk 1 above)
Comment (aleixmorgadas) [reply src/model/account/PublicAccount.ts]: Yes, does it make sense? Or should we thrown an exception?

File: test/model/account/Account.spec.ts (hunk 4 above)
Comment (aleixmorgadas) [reply test/model/account/Account.spec.ts]: `to.be.true` is an issue with tslint, we should 

a) change the tslint rules
b) use `equal(true)`.

I like `to.be.true`

Comment (Vektrat) [reply test/model/account/Account.spec.ts]: I prefer to accept `to.be.true` as well, but np it's a detail

**Pull request merged**